### ISSUE TITLE
View Logs and Execute Shell do not work when tried on workloads

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#18466]
**What kind of request is this:** bug

**Steps to reproduce:**
1. Launch Rancher. Create DO/GKE clusters
2. Create Workloads
3. Click on the workload and try to 'Execute Shell' or 'View Logs'

**Expected Result:**
User should be able to view the logs or exec into the pod successfully.

**Actual Result:**
On clicking on 'Execute Shell', the window pops up and closes immediately.

Request response:
<img width="540" alt="screen shot 2019-02-27 at 4 45 49 pm" src="https://user-images.githubusercontent.com/26032343/53533534-1e637380-3ab0-11e9-8b7b-8bc48f8a360a.png">

On clicking on View Logs, status is shown as `Disconnected`
<img width="1639" alt="screen shot 2019-02-27 at 4 48 05 pm" src="https://user-images.githubusercontent.com/26032343/53533373-806fa900-3aaf-11e9-91d1-5c5ce694da77.png">

**Other details that may be helpful:**

**Environment information**
- Rancher version: v2.1.7-rc4
- Installation option (single install/HA): Single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): DO/GKE
- Kubernetes version (use `kubectl version`): 

```
v1.12.6/v1.13.1/v1.11.7-gke.6

```

- Docker version (use `docker version`):

```
18.9.2
```

**Comment by sangeethah** [comment]
Not able to launch kubectl shell for UI for DO/GKE cluster as well.

Also not able to exec into pods using remote kubectl access as well.
```
kubectl exec -it testclient-6749d887cc-fpnrk bash
error: Timeout occured
```

**Comment by alena1108** [comment]
Fixed in `rancher/rancher:v2.1.7-rc7`

**Comment by sowmyav27** [comment]
verified in `rancher/rancher:v2.1.7-rc7`. 
View Logs and Execute Shell work. 

<img width="545" alt="screen shot 2019-03-01 at 11 53 25 am" src="https://user-images.githubusercontent.com/26032343/53662884-fa6f7180-3c18-11e9-91c8-71aa77e8e66b.png">

<img width="1609" alt="screen shot 2019-03-01 at 11 54 23 am" src="https://user-images.githubusercontent.com/26032343/53662885-fc393500-3c18-11e9-8a5d-348519aaa975.png">
